### ISSUE TITLE
[BREAKING] bug

### ISSUE THREAD

**Issue by mercedes7935** [sk7725/TimeControl#39]
I am running v7 and i noticed that a lot of things run into problems, if i speed up. for example, when i go to 8 times speed (lower same issue but slower) my power grid failed multiple times now. The reason for me is that i need power to make more power and when i lose my cryptofluid my main generators go offline and i have to do manually start up. the issue is that the crypto fluid isnt working correctly. during speed-ups the tanks gets slowly emptied and as soon as i reset to 1x it fills up again pretty quickly. i saw some other things which look to me like the calculation or generally the speed-up is not working properly. This is really sad and makes the mod unusable!

**Comment by MrChampion1234** [comment]
I'm having this problem too, only I first noticed it happens with the water for Steam Generators. When I sped up the game, the water levels in the conduits would mysteriously go down, and then jump back up again when I reset the speed.

So, I decided to look into it a bit further to understand why this is happening. I'm not a mod developer, nor do I know much about how the game works technically, but, from what I can tell, a lot of the game's mechanics are limited by frame rate. If you want a tldr on how to fix it, you'll basically need to get as high of an FPS as possible (like above 200) because for some reason the game's logic is tied to the frame rate. So, turn off VSync, bloom, linear filtering, etc, anything that could slow it down you want to turn off, Settings -> Graphics is where you can find those settings. Then, also in Settings -> Graphics, drag the Max FPS all the way to the right until it says "None". You may also want to turn on "Show FPS & Ping" to see how high the FPS is. This probably won't work for all speeds, but it'll work perfectly for some of them.

If you don't believe me, you can try this out by going into sandbox mode, and using this schematic [distributortest.zip](https://github.com/sk7725/TimeControl/files/10808766/distributortest.zip), then changing the speed to something like 64x, you'll notice that the items are getting blocked by the distributor. See how low it needs to be before items can get through (for me it's 4x speed). Now, turn VSync off (this would sync the framerate with your monitor's framerate, so turn that off), and drag the Max FPS slider all the way to the right, and now go back and try setting it to whatever speed caused it to break before, and, hopefully, you should find that it works just fine now. Chances are, you're not going to get it up to, say, 256x speed, or even above 16x speed, without it breaking, but with this you might be able to get it up to 8x speed. 

I hope this helps, since it certainly did for me.

**Comment by Raif1** [comment]
Also, units automatically die.

**Comment by Penrosian** [comment]
The units dying thing is kinda weird but it happened to me too, and the FPS thing got me from x8 to x64 speed working.

**Comment by KartikVerma29** [comment]
Yes, I also suffered from the same issue regarding Power losses when speed up upto x4 or more

**Comment by sk7725** [comment]
The liquid physics in this game is not frame-dependent. In other words, even if the game is 0.5x speed or 2x speed, liquid is still stuck at x1. This has little problem on lower speeds because liquid is almost instant anyways, but has a problem at higher speeds. The mod warns you about this at higher speeds.

This is technically a Mindustry issue and impossble for a mod to fix.

**Comment by KartikVerma29** [comment]
so, if we use power generators without Liquid coolants, it won't break. Right ?

**Comment by sk7725** [comment]
There may be other features that may not scale with time, but as of v7 liquids and multiplayer servers are the most notable ones. Cannot say for v8 though.
